### ISSUE TITLE
Escape HTML syntax in docstring rendering

### ISSUE THREAD

**Issue by klonuo** [astral-sh/ty#3482]
### Summary

Hi,

I first reported the issue here: https://github.com/sublimelsp/LSP/issues/2915

It seems that html tags should be escaped by the server for the hover

### Version

ty 0.0.37 (f18aed643 2026-05-16)

**Comment by MichaReiser** [comment]
Makes sense

`main.py`:

```py
def urlparse(url, scheme='', allow_fragments=True):
    """Parse a URL into 6 components:
    <scheme>://<netloc>/<path>;<params>?<query>#<fragment>

    The result is a named 6-tuple with fields corresponding to the
    above. It is either a ParseResult or ParseResultBytes object,
    depending on the type of the url parameter.

    The username, password, hostname, and port sub-components of netloc
    can also be accessed as attributes of the returned object.

    The scheme argument provides the default value of the scheme
    component when no scheme is found in url.

    If allow_fragments is False, no attempt is made to separate the
    fragment component from the previous component, which can be either
    path or query.

    Note that % escapes are not expanded.

    urlsplit() should generally be used instead of urlparse().
    """
    ...

urlparse("test")
```

`ty.json`:

```json
{
    "environment": {
        "python-version": "3.14"
    },
    "rules": {
        "undefined-reveal": "ignore"
    }
}
```

[Playground](https://play.ty.dev/a8470d3a-7cf5-482b-afa7-d786cb72a7f2)
